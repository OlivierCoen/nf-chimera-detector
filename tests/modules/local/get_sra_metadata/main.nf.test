nextflow_process {

    name "Test Process GET_SRA_METADATA"
    script "modules/local/get_sra_metadata/main.nf"
    process "GET_SRA_METADATA"
    tag "get_sra_metadata"

    test("Too long request URI") {

        when {

            process {
                """
                input[0] = [
                    [ family: "test" ],
                    10244
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
