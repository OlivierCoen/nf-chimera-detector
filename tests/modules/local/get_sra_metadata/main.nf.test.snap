{
    "Too long request URI": {
        "content": [
            {
                "0": [
                    [
                        {
                            "family": "test"
                        },
                        "10244.sra_ids.txt:md5,30e57aa0ec169aaf9c56ebfb2c926455"
                    ]
                ],
                "1": [
                    [
                        {
                            "family": "test"
                        },
                        "10244.sra_metadata.json:md5,e672c38d6ee360e07f23825d5ab1ff91"
                    ]
                ],
                "2": [
                    [
                        "GET_SRA_METADATA",
                        "python",
                        "3.13.5"
                    ]
                ],
                "3": [
                    [
                        "GET_SRA_METADATA",
                        "requests",
                        "2.32.4"
                    ]
                ],
                "4": [
                    [
                        "GET_SRA_METADATA",
                        "xmltodict",
                        "0.14.2"
                    ]
                ],
                "5": [
                    [
                        "GET_SRA_METADATA",
                        "tenacity",
                        "9.1.2"
                    ]
                ],
                "sra_id_files": [
                    [
                        {
                            "family": "test"
                        },
                        "10244.sra_ids.txt:md5,30e57aa0ec169aaf9c56ebfb2c926455"
                    ]
                ],
                "sra_metadata": [
                    [
                        {
                            "family": "test"
                        },
                        "10244.sra_metadata.json:md5,e672c38d6ee360e07f23825d5ab1ff91"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-11T17:53:40.013883873"
    }
}